src/iot-setup.py

```diff
@@ -0,0 +1,348 @@
+            print("No deployments found to cancel")
+        else:
+            for deployment in deployments['effectiveDeployments']:
+                gg_client.cancel_deployment(
+                    deploymentId=deployment['deploymentId'])
+            print(f"Cancelled all deployments")
+
+    except Exception as e:
+        print(f"Error when listing deployments: {str(e)}")
+        return False
+
+    # Delete the core device
+    try:
+        gg_client.delete_core_device(coreDeviceThingName=thing_name)
+        print(f"Deleted Greengrass core device '{thing_name}'")
+    except Exception as e:
+        print(f"Failed to delete Greengrass core device: {str(e)}")
+        return False
+
+    return True
+
+
+def delete_thing(thing_name: str, iot_client: client):
+    try:
+        # Get certificates that attached to the thing
+        principals = iot_client.list_thing_principals(
+            thingName=thing_name)['principals']
+
+        # For each certificate attached to the thing
+        for principal in principals:
+            cert_id = principal.split('/')[-1]
+
+            # Detach all policies from the certificate
+            policies = iot_client.list_principal_policies(
+                principal=principal)['policies']
+
+            for policy in policies:
+                iot_client.detach_policy(policyName=policy['policyName'],
+                                         target=principal)
+
+            # Detach certificate from thing
+            iot_client.detach_thing_principal(thingName=thing_name,
+                                              principal=principal)
+
+            # Update certificate to INACTIVE
+            iot_client.update_certificate(certificateId=cert_id,
+                                          newStatus='INACTIVE')
+
+            # Delete the certificate
+            iot_client.delete_certificate(certificateId=cert_id,
+                                          forceDelete=True)
+
+        # Finally, delete the thing
+        iot_client.delete_thing(thingName=thing_name)
+
+        print(
+            f"Successfully deleted thing '{thing_name}' and its associated certificates"
+        )
+        return True
+
+    except ClientError as e:
+        error_code = e.response['Error']['Code']
+        error_message = e.response['Error']['Message']
+        print(
+            f"Error deleting thing '{thing_name}': {error_code} - {error_message}"
+        )
+        return False
+    except Exception as e:
+        print(f"Unexpected error deleting thing '{thing_name}': {str(e)}")
+        return False
+
+
+def clean_up(region: str, thing_group_name: str):
+    iot_client = client("iot", region_name=region)
+    gg_client = client("greengrassv2", region_name=region)
+    remove_all_things_from_thing_group(thing_group_name, iot_client, gg_client)
+    iot_client.delete_thing_group(thingGroupName=thing_group_name)
+
+    try:
+        subprocess.run(['sudo', 'rm', '-rf', JSON_FILE], check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)